File: Omkar's Pet Cat/claude1.py
import sys
from math import gcd

def solve(A):
    g = 0
    for x in A:
        g = gcd(g, x)
    cnt = A.count(g)
    return "NO" if cnt >= 2 else "YES"

def main():
    data = sys.stdin.read().split()
    idx = 0
    T = int(data[idx]); idx += 1
    out = []
    for _ in range(T):
        N = int(data[idx]); idx += 1
        A = list(map(int, data[idx:idx+N])); idx += N
        out.append(solve(A))
    sys.stdout.write("\n".join(out) + "\n")

main()

File: Omkar's Pet Cat/claude1.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int T;
    scanf("%d", &T);
    while(T--){
        int N;
        scanf("%d", &N);
        vector<int> A(N);
        for(auto &x : A) scanf("%d", &x);
        int g = A[0];
        for(int i = 1; i < N; i++) g = __gcd(g, A[i]);
        int cnt = 0;
        for(int x : A) if(x == g) cnt++;
        puts(cnt >= 2 ? "NO" : "YES");
    }
    return 0;
}